spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/EventHubTemplate.java

```diff
@@ -57,12 +57,13 @@ public EventHubTemplate(EventHubClientFactory clientFactory) {
         Assert.hasText(eventHubName, "eventHubName can't be null or empty");
         Assert.notNull(message, "message can't be null");
         try {
-            EventHubClient client = this.clientFactory.getOrCreateEventHubClient(eventHubName);
+            EventHubClient client = this.clientFactory.getEventHubClientCreator().apply(eventHubName);
```

Comment:
Wondering why we'd introduce this Memoizer and Function? I indeed think clientFactory.getOrCreateEventHubClient is better than clientFactory.getEventHubClientCreator().apply.

Reply:
Memoization is used to cache expensive resources. EventHubClient isn't cacheable as it is tightly coupled with connectionstring and eventhub name. A new eventHubName means a new creation.
https://www.javatips.net/api/commons-lang-master/src/main/java/org/apache/commons/lang3/concurrent/Memoizer.java

Reply:
1. Memoizer is used to remove duplication of `putIfAbsent` operation of map. Another benefit is to separate the concern of different function by taking adavantage of functional composition.
2. EventHubClient is only coupled by eventhub name. So we can use Memoizer to cache with name as input and client as output.

Reply:
Good learning to me. Thanks.